### DISCUSSION TITLE
Autofocus

### DISCUSSION THREAD

**Discussion by hosken79** [raspberrypi/picamera2#181] [General]
Hi,

Is there there a way to set Autofocus on a camera ? I've looked and i cant see an example for picamera2

Thanks for help in advance.

**Comment by davidplowman** [comment]
Hi, there is no support in Raspberry Pi's libcamera implementation for autofocus, and therefore it isn't supported in Picamera2 either. Some 3rd parties have implemented some versions of autofocus for their own modules, but you would have to get in touch with them. Autofocus is on our list of things to look into, but I can't give you a timescale, I'm afraid.
